internal/adminapi/endpoints.go

```diff
@@ -0,0 +1,76 @@
+package adminapi
+
+import (
+	"context"
+	"fmt"
+
+	discoveryv1 "k8s.io/api/discovery/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetURLsForService performs an endpoint lookup, using provided kubeClient
+// to list provided service's endpointslices.
+func GetURLsForService(ctx context.Context, kubeClient client.Client, service types.NamespacedName) ([]string, error) {
+	// Get all the endpointslices assigned to the provided service.
+	labelReq, err := labels.NewRequirement("kubernetes.io/service-name", selection.Equals, []string{service.Name})
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		addresses     []string
+		continueToken string
+	)
+	for {
+		var endpointsList discoveryv1.EndpointSliceList
+		if err := kubeClient.List(ctx, &endpointsList, &client.ListOptions{
+			LabelSelector: labels.NewSelector().Add(*labelReq),
+			Namespace:     service.Namespace,
+			Continue:      continueToken,
```

Comment:
What the Continue is needed for without having a Limit set? Isn't it used to get incremental chunks of elements instead the whole set at once?

Reply:
Continue is used as a token for paging:

> 	// Limit specifies the maximum number of results to return from the server. The server may
> 	// not support this field on all resource types, but if it does and more results remain it
> 	// will set the continue field on the returned list object. This field is not supported if watch
> 	// is true in the Raw ListOptions.
> 	Limit int64
> 	// Continue is a token returned by the server that lets a client retrieve chunks of results
> 	// from the server by specifying limit. The server may reject requests for continuation tokens
> 	// it does not recognize and will return a 410 error if the token can no longer be used because
> 	// it has expired. This field is not supported if watch is true in the Raw ListOptions.
> 	Continue string

But you're right I think. Without the limit set here, there would not be any other limit set by the controller-runtime/api to force paging on big requests I believe.

We can set a "sane" limit here to force paging to prevent big responses chocking the api server.

Reply:
Set a default limit to 100 and put it in a `const`.